docs/007-funding-rate.adoc

```diff
@@ -0,0 +1,96 @@
+= ADR 007 - Funding rate
+Lucas Soriano <lucas@10101.finance>
+:toc:
+:icons: font
+:attributes: 2023-09-06
+
+Status: Proposed
+
+== Decision
+
+The funding rate should be charged through a Lightning payment.
+The funding rate *must* be paid before renewing the contract.
+No funding rate needs to paid for the period of time _before_ the first renewal.
+
+== Context
+
+The funding rate is the cost of holding a perpetual swap position that is on the side of the market sentiment.
+There needs to be a funding rate because that is what motivates traders to keep their perpetual swaps positions open _against_ the market sentiment.
+The sign of the funding rate determines which of the two parties involved must pay the funding fee to keep their position open.
+
+In the 10101 ecosystem, the funding rate must come from the 10101 orderbook.
+This value will be updated at a fixed interval e.g. every 8 hours.
+
+== Options considered
+
+=== Lightning payment before the DLC is renewed
+
+The renewal protocol will first require one of the two parties (coordinator and trader) to generate an invoice for the funding fee.
+Which party generates the invoice (i.e. the receiver) is determined by the sign of the funding rate of the 10101 orderbook.
+
+Only after the invoice has been paid will the coordinator initiate the renew protocol to roll over the DLC between coordinator and trader.
+
+==== Pros
+
+- The DLC remains unaffected by the exchanged funding rate.
+- Can be implemented without changes to our dependencies.
+- Easy to track the funding fee payments.
+- Fee is readily available to use after.
+
+==== Cons
+
+===== No execution guarantees
+
+There is no guarantee that the renew protocol will succeed after the funding fee payment is claimed.
+Depending on the size of the position, it could be economically beneficial for a party to claim the funding fee paid by the other party and then force-close the channel.
+As such, in some scenarios we would be trusting the counterparty to not run away with an unearned funding fee.
+
+With PTLCs we might be able to make the DLC channel update and the payment atomic, but that is not yet implemented.
+
+===== Funds required on Lightning channel
+
+To be able to pay the funding fee the payer will need to have sufficient outbound liquidity.
+It might happen that a user will need to stop trading if their Lightning channel cannot cover the costs.
+
+They would still be able to close their position and open a new one.
+
+=== Adjust the CET payouts depending on the funding fee
+
+If the funding fee is `X` paid from `A` to `B`, for every `o_A`, CET output amount belonging to `A`, we would subtract `X`: `o_A - X`; and for every `o_B`, CET output belonging to `B`, we would add `X`: `o_B + X`.
+
+Unfortunately this doesn't work in all scenarios:
+
+- If a CET already only pays to `B` then the payout cannot be increased!
```

Comment:
If this is the case the user (`A`) was basically liquidated.

Reply:
This is true. But that does not mean that the funding fees were paid.

Basic example:

- 1st rollover makes `A` pay to `B`. The "A gets liquidated" CET cannot get adjusted because the output already pays everything to `B`.
- At the updated expiry the oracle implicitly unlocks the "A gets liquidated" CET. `B` should have earned the full DLC payout _plus the funding fee_, but they only get the DLC payout.

In a more extreme scenario, if the contract is rolled over 20 times and the funding fee is paid 15 times from `A` to `B`, if `A` gets liquidated with the CET at the boundary of the curve, `B` misses out on `15 - 5 = 10` trading fees.

Obviously this only needs to apply to contracts that are closed on-chain. If I were keeping track of the funding fees outside of the CETs I would want to do the accounting without capping gains/losses based on the collateral. But this would mean there would be a discrepancy between on-chain and off-chain settlement.